lib/datadog/appsec/processor.rb

```diff
@@ -94,6 +80,30 @@ def finalize
 
       private
 
+      def update_rules_data_with_static_configured_values
+        ruledata_setting = []
+        ruledata_setting << ip_denylist_data(Datadog::AppSec.settings.ip_denylist || [])
+        ruledata_setting << user_id_denylist_data(Datadog::AppSec.settings.user_id_denylist || [])
+
+        update_rule_data(ruledata_setting)
+      end
+
+      def ip_denylist_data(denylist, id: 'blocked_ips')
+        {
+          'id' => id,
+          'type' => 'data_with_expiration',
+          'data' => denylist.map { |ip| { 'value' => ip.to_s, 'expiration' => 2**63 } }
+        }
+      end
+
+      def user_id_denylist_data(denylist, id: 'blocked_users')
+        {
+          'id' => id,
```

Comment:
With the current implementation, the `id:` kwarg must always be the same value, so it should be folded in (and probably turned into a constant):

```suggestion
      def user_id_denylist_data(denylist)
        {
          'id' =>  'blocked_users',
```

But in truth I'm thinking both this and the note below pointing at becoming a single `denylist_data(denylist, id)` method. WDYT?